src/nonLinPMMH.cpp

```diff
@@ -0,0 +1,146 @@
+// Gordon et al. (1993) and Kitagawa (1996).
+//
+// Copyright (C) 2017         Dirk Eddelbuettel, Adam Johansen and Leah South
+//
+// This file is part of RcppSMC.
+//
+// RcppSMC is free software: you can redistribute it and/or modify it
+// under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 2 of the License, or
+// (at your option) any later version.
+//
+// RcppSMC is distributed in the hope that it will be useful, but
+// WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with RcppSMC.  If not, see <http://www.gnu.org/licenses/>.
+
+#include "nonLinPMMH.h"
+
+namespace nonLinPMMH {
+    const double a_prior = 0.01;
+	const double b_prior = 0.01;
+}
+
+using namespace std;
+using namespace nonLinPMMH;
+
+
+// nonLinPMMH_impl() function callable from R via Rcpp::
+// [[Rcpp::export]]
+Rcpp::DataFrame nonLinPMMH_impl(arma::vec data, unsigned long lNumber, unsigned long lMCMCits) { 	
+  
+  try {
+	arma::vec sigv(lMCMCits), sigw(lMCMCits);
+	long lIterates = data.n_rows;
+    y = data;
+	
+	arma::vec loglike = arma::zeros(lMCMCits);
+	arma::vec logprior = arma::zeros(lMCMCits);
+	
+	double dRand;
+	double loglike_prop;
+	double logprior_prop;
+	
+    //Initialise and run the sampler
+    smc::sampler<double> Sampler(lNumber, HistoryType::NONE);
+	theta_prop.sigv = 10.0;
+	theta_prop.sigw = 10.0;
+		
+	sigv(0) = theta_prop.sigv;
+	sigw(0) = theta_prop.sigw;
+	
+	// Getting a particle filtering estimate of the log likelihood.
+	smc::moveset<double> Moveset(fInitialise, fMove, NULL);
+    Sampler.SetResampleParams(ResampleType::MULTINOMIAL, 0.5);
+    Sampler.SetMoveSet(Moveset);
+    Sampler.Initialise();
+	Sampler.IterateUntil(lIterates-1);
+	loglike(0) = Sampler.GetLogNCPath();
+	
+	// Inverse gamma prior
+	logprior(0) = logPrior(theta_prop);
+    
+	double MH_ratio;
+	for (unsigned int i = 1; i<lMCMCits; i++){
+		// RW proposal for parameters
+		theta_prop.sigv = R::rnorm(sigv(i-1),0.15);
+		theta_prop.sigw = R::rnorm(sigw(i-1),0.08);		
+		
+		// Getting a particle filtering estimate of the log likelihood.
+		Sampler.Initialise();
+		Sampler.IterateUntil(lIterates-1);
+		loglike_prop = Sampler.GetLogNCPath();
+		
+		// Inverse gamma prior
+		logprior_prop = logPrior(theta_prop);
+		
+		MH_ratio = exp(loglike_prop - loglike(i-1) + logprior_prop - logprior(i-1));
+		dRand = R::runif(0,1);
+      
+		if (MH_ratio>dRand){
+			sigv(i) = theta_prop.sigv;
+			sigw(i) = theta_prop.sigw;
+			loglike(i) = loglike_prop;
+			logprior(i) = logprior_prop;
+			Rcpp::Rcout  << "ACCEPTED proposal (" << setw(6) << sigv(i) << ", " << setw(6) <<  sigw(i) << ") at iteration " << setw(4) << i+1 << " with loglike " << setw(8) << loglike(i) << std::endl;
```

Comment:
Fair enough. I wouldn't personally want ~10k lines of output printed to the console when doing something like this; a boolean "verbose" or "quiet" argument might be a good idea so that the user can suppress it. Personally, I'd just print an "iteration xx/yy" every 100 or 1000 iterations if I wanted to reassure the user that something was happening.